Update dependencies

djc opened the pull request:
Would be nice to get this in a release.

Given the lack of activity, what will maintenance be like going forward?

djc committed:
Update ordered-float to version 4

djc committed:
Update itertools to 0.11

djc committed:
Update memmap2 to 0.9

djc committed:
Update approx to 0.5

djc committed:
Upgrade criterion to 0.5

djc committed:
Upgrade to toml 0.8

sebpuetz commented:
> Given the lack of activity, what will maintenance be like going forward?

I'm happy to review changes / dependency updates, I haven't been using the library (or rather anything in the project) for a couple of years so nothing has been driving active development.

I'll try to go through the other repos during the week and take care of some updates.

djc commented:
Thanks for the quick response!

djc commented:
Looks like the mips64 target is not available for Rust 1.72: https://github.com/rust-lang/rust/issues/115218.

Let me know how you want this handled?

sebpuetz commented:
> Looks like the mips64 target is not available for Rust 1.72: [rust-lang/rust#115218](https://github.com/rust-lang/rust/issues/115218).
> 
> Let me know how you want this handled?

After a cursory look at the issue, it looks like RustCrypto is using PPC32 as an alternate BE target: https://github.com/RustCrypto/sponges/pull/62

I believe our intention was to test BE targets, too, so that should be a viable alternative

djc committed:
Bump MSRV to 1.66

djc committed:
Add rust-version metadata

djc committed:
Apply clippy suggestions

djc committed:
Use powerpc-unknown-linux-gnu to test big-endian

The mips64 target we used before has been demoted to tier 3:

https://github.com/rust-lang/rust/issues/115218

djc committed:
Explicitly ignore Cargo.lock

djc committed:
Remove non-existent file from package.exclude

djc committed:
Remove authors from Cargo metadata (per RFC 3052)

djc committed:
Upgrade to 2021 edition

djc committed:
Bump version to 0.18.0

djc commented:
I added a version bump to 0.18, since some of the dependencies that got bumped are visible in the public API.

danieldk commented:
> I'm happy to review changes / dependency updates, I haven't been using the library (or rather anything in the project) for a couple of years so nothing has been driving active development.

Same here, I don't use finalfusion/finalfrontier anymore, but am happy to review stuf

danieldk commented on src/chunks/metadata.rs:
```diff
@@ -16,18 +15,18 @@ use crate::io::ReadMetadata;
 #[derive(Clone, Debug, PartialEq)]
 pub struct Metadata {
-    inner: Value,
+    inner: toml::Table,
```
Any reason not to import `toml::Table` above?

djc replied:
Changed it to import the type.

danieldk commented on src/compat/fasttext/io.rs:
```diff
@@ -107,7 +105,7 @@ impl ReadFastTextPrivate for Embeddings<FastTextSubwordVocab, NdArray> {
         }
 
-        let metadata = Value::try_from(config).map_err(|e| {
+        let metadata = toml::Table::try_from(config).map_err(|e| {
```
Same.

sebpuetz reviewed (approved):
LGTM

djc commented:
Thanks! Are there any other blockers to getting this published on crates.io?

danieldk commented:
> Thanks! Are there any other blockers to getting this published on crates.io?

Nope, I'll get to it now.

danieldk commented:
Done, thanks for making this happen!
